### ISSUE TITLE
Open Playwright trace from HTML report without downloading

### ISSUE THREAD

**Issue by vitalets** [vitalets/playwright-bdd#103]
Hi @vitalets, the html report looks great. 
One thing I wanted to ask if you can make it possible the traces of a failed test to be opened directly within the browser with a click. Currently I need to download the trace file and open it with command 'npx playwright show-trace ....'.

Within TeamCity when I am using the playwright html report I also have the option to just click it and it will open in a frame, so no need to download it. 

![image](https://github.com/vitalets/playwright-bdd/assets/94455079/154a6673-b277-4de4-9dc6-22a1ed436472)

![image](https://github.com/vitalets/playwright-bdd/assets/94455079/0195e55d-5c1f-4686-bba1-2acfa244e5ba)

this is sample from the html how it is opened in TeamCity iframe

![image](https://github.com/vitalets/playwright-bdd/assets/94455079/6d13e1c6-8523-485b-9665-2e3e7594ca52)

Maybe passing the URL of the traces could help 

![image](https://github.com/vitalets/playwright-bdd/assets/94455079/6a18bf1a-8a99-4f4c-99a5-b946e5b564cb)

_Originally posted by @NikkTod in https://github.com/vitalets/playwright-bdd/issues/9#issuecomment-1983940651_

**Comment by vitalets** [comment]
Interesting idea, should think about it. 
The main point here is that `trace.zip` should be uploaded to some public storage to be opened with `trace.playwright.dev`.

There are two questions:
1. Could you check, that artefacts from your teamcity server are available to `trace.playwright.dev`? I mean checking that url like this `https://trace.playwright.dev/?trace=https://you-teamcity-server/../trace.zip` shows trace?
2. if you use Playwright's original HTML reporter, do you still need to download trace.zip to open it? If yes, do you use `attachmentsBaseURL` option?

**Comment by NikkTod** [comment]
Hi @vitalets, we should have a look at how playwright team are making the remote view traces available directly within their html report. 

![image](https://github.com/vitalets/playwright-bdd/assets/94455079/d0eeb723-49f8-442e-8dd3-6f4a737cf63e)

I had a look in TeamCity on our side and this is what I see:

The view option in the playwright's original HTML reporter generates the following link: 

`https://our-teamcity-server:id/reports/trace/index.html?trace=https://our-teamcity-server:id/reports/data/traceID.zip`

If I copy and paste this link in a new browser tab, the traces are opened directly within that tab. 

Just for your information the only thing that I am configuring in TeamCity is to point to the path where the html playwright report is being generated. 

If I open in a browser tab only the second part of the above link it will just download the zip file -> `https://our-teamcity-server:id/reports/data/traceID.zip`

So it seems that they do not use in their html report view the `https://trace.playwright.dev/` url, but the following structure `https://our-teamcity-server:id/reports/trace/index.html?trace=.....`

So from all above observations let me try to answer the questions:

1. I am not able to open the traces if I use the `trace.playwright.dev` url. It gives me an error 
`"Could not load trace from https://our-teamcity-server:id/reports/data/traceID.zip" Make sure a valid Playwright Trace is accessible over this URL."`  But if I paste that `https://our-teamcity-server:id/reports/data/traceID.zip` in a browser tab the zip is being downloaded successfully.

2. When using the Playwright's original HTML reporter I do not download the traces zip, I directly open the traces from the html report. I am not using `attachmentsBaseURL `

**Comment by vitalets** [comment]
Nice investigation. I didn't realize that Playwright HTML report also acts as trace viewer.
I've check Playwright HTML reporter source, they generate trace url [this way](https://github.com/microsoft/playwright/blob/7f6f17d1c343f88400c498fd1c3defbd7f420056/packages/html-reporter/src/links.tsx#L88):
```ts
export function generateTraceUrl(traces: TestAttachment[]) {
  return `trace/index.html?${traces.map((a, i) => `trace=${new URL(a.path!, window.location.href)}`).join('&')}`;
}
```
What we can do for Cucumber reports:
1. output Playwright HTML report along with Cucumber HTML report
2. generate trace links in Cucumber report redirecting to Playwright HTML report

Another option is to embed Playwright [trace viewer](https://github.com/microsoft/playwright/tree/main/packages/trace-viewer) into Cucumber report somehow.

One more thing I've noticed - traceviewer works only on `http / https` urls, on `file://` scheme it shows an error:
![image](https://github.com/vitalets/playwright-bdd/assets/1473072/16d5dc19-934d-43da-ba8f-b762d45d544f)
For Cucumber reports on Teamcity it will be ok (as they are on `https://`, for opening report files locally it will not work.

Lets think more about it.

**Comment by NikkTod** [comment]
Embeding the Playwright trace-viewer into Cucumber report sounds great.

As you said, currently you can not open the traces from the playwright html report locally, it gives the above error, but I don't think this is a stopper, because locally most of the users are using directly `trace.playwright.dev`.

In my case, because of company restrictions, the `trace.playwright.dev` url is blocked, so if needed locally I am using the 'npx playwright show-trace path/to/trace.zip'

Maybe we can create a feature ticket on playwright side to allow also `file://`

**Comment by arieluchka-bse** [comment]
adding a flag to create an "All-in-one" tracer zip would also be amazing! :)
being able to send the tracer zip and knowing that the receiving end can watch it without installing playwright/going to trace.playwright.dev

**Comment by vitalets** [comment]
@arieluchka-bse Just to clarify, under 'all-in-one' tracer you mean Cucumber report bound with tracer, right?
So you can generate cucumber report in some directory, zip this directory and send to someone?

**Comment by arieluchka-bse** [comment]
hi, apologies if it's the wrong repo to comment on.
im fairly new to playwright and testing in general, and i've never heard about cucumber report.
im using playwright with python, so for test reporting im using pytest.
what i meant by "all-in-one" is just for the tracer.
generating a tracer.zip with all html files that are needed to see the trace, without depending on the [trace.playwright.dev](https://trace.playwright.dev/).

from what i saw, in NodeJS there is an option to set the "reporter" to html, and all the test results + **traces**, can be opened locally with the generated reporter (so no need to navigate to the site.

i wanted to implement something similar to what's showing [here](https://www.veed.io/view/8a152ac1-d9c0-420f-ba65-187a8fe54688?panel=share), with providing links to trace.zips, but the author of this achived it with setting the reporter to html ([tutorial](https://afsalbacker.medium.com/access-playwright-trace-viewer-reports-online-via-amazon-s3-51fd365d80f6)).

just wish there was something like this for python, that will allow viewing traces, just by opening the zip file (without downloading or navigating anywhere else)

**Comment by vitalets** [comment]
There is Playwright's CLI command [show-trace](https://playwright.dev/docs/trace-viewer#opening-the-trace), does it serve your needs?
```
npx playwright show-trace path/to/trace.zip
```

**Comment by arieluchka-bse** [comment]
hi, 
what i meant is being able to share traces with co-workers that do not have playwright installed.

in Node, if we set reporter=html, the test results are outputed into a zip that is accessed as an html file.

this makes it very easy to send the results to people who do not have playwright installed (they can just open the html file, and see all test results and traces, **without** needing https://trace.playwright.dev/ to view the trace, and **without** installing playwright locally).

This functionality exists in node, but not in python :(

**Comment by vitalets** [comment]
@arieluchka-bse I agree such case is useful.
One note here: to see the trace people need to open html report via https, not just clicking to index.html and open on file:// protocol. 
Anyway, embedding trace-viewer into cucumber report is good feature and is planned in playwright-bdd.

**Comment by vitalets** [comment]
Trace view in Cucumber HTML report is released in Playwright-bdd v7.5.0.
Feel free to open new issues in case of any errors.
